### PR TITLE
Update gems

### PR DESCRIPTION
**Pull request by kevintyll**
Update gems to current versions.
Add bundler-audit to travis run.
Fix new rubocop offenses.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,10 +6,13 @@ rvm:
   - 2.1
   - 2.2
   - 2.3.0
+  - 2.4.0
 script:
   - bundle exec rake test
   - bundle exec rake coveralls:push
```
Comment (johnathanludwig) [inline .travis.yml]: We should probably remove coveralls. We haven't really been paying attention to it, and it causes errors when run internally.

File: lib/esp/extensions/active_resource/formats/json_api_format.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ def decode(json)
         Formats.remove_root(parse_json_api(ActiveSupport::JSON.decode(json)))
       end
 
-      private
+      private_class_method
```
Comment (johnathanludwig) [inline lib/esp/extensions/active_resource/formats/json_api_format.rb]: Is this doing what you expect it to? I believe you have to pass the method names to `private_class_method`.

Comment (kevintyll) [inline lib/esp/extensions/active_resource/formats/json_api_format.rb]: Good question.   It shuts rubocop up, which is what I want.

Comment (kevintyll) [inline lib/esp/extensions/active_resource/formats/json_api_format.rb]: Apparently it does not do what I want.  Changing everywhere I've done that over the last 4 months now.

File: .travis.yml (hunk 1 above)
Comment (kevintyll) [inline .travis.yml]: Done

**Pull request merged**